### DISCUSSION TITLE
Condição para midi ser enviado ou não

### DISCUSSION THREAD

**Discussion by Willame156** [tttapa/Control-Surface#649] [General]
Em primeiro lugar, obrigado por esta ótima biblioteca.
Gostaria que o button_01 só enviasse mensagem midi CC se determinada condição ocorresse, mas não consegui.

```cpp
# include  < Control_Surface.h >     
# include  < MIDI_Interfaces / BluetoothMIDI_Interface.hpp > 
BluetoothMIDI_Interface midi;  

if (digitalRead (botton_02) == HIGH); {    // Condição para o button_01 enviar midi
CCButtonLatched button_01 = { 23 ,  { 4 , CHANNEL_1},  };     // Não leia "button_01" se button_02 tiver pressionado
} else { 
                
void  setup () {
   Control_Surface. begin ();
}

void  loop () {
  Control_Surface. loop ();
}
```
Agradeço a sua atenção

**Comment by tttapa** [comment]
You cannot do this using the standard `CCButtonLatched` class like that. You'll have to program the logic yourself as demonstrated in https://tttapa.github.io/Control-Surface-doc/Doxygen/d4/d3b/Custom-MIDI-Output-Element_8ino-example.html.

Note that you cannot have top-level if statements like that, it has to be inside of a function. Don't terminate the if statement using a semicolon, you need a block statement (`{ ... }`) directly after `if (...)`. See https://www.learncpp.com/cpp-tutorial/introduction-to-if-statements/ for more details.

**Reply by Willame156** [reply]
// Feliz Natal tttapa!
// Muito obrigado pela resposta rápida.
// Não consegui fazer igual ao exemplo que você me indicou, mas a partir dele e de outros que vi consegui fazer.
// Talvez haja uma maneira melhor de fazer o mesmo resultado, mas só consegui assim.
// Não sou programador, mas com sua biblioteca e esforço vejo que é possível criar algo interessante.

 #include <Control_Surface.h>    // Inclua a biblioteca Control Surface
#include <MIDI_Interfaces/BluetoothMIDI_Interface.hpp> //Inclui a função de bluetooth

BluetoothMIDI_Interface midi;  //Habilita a conexão Bluetooth.


Button pushbutton_TAP {0}; //Botão conectado ao pino 0 do ESP 32.

int proximity=0;

  
void setup() {
  pushbutton_TAP.begin(); // enables internal pull-up
  midi.begin();       // initialize the MIDI interface
}
 
void loop() {
proximity = 900; // Simula leitura de um sensor de proximidade.
  
  midi.update(); // read and handle or discard MIDI input
if(proximity != 1023) {                  //Condição para o botão funcionar
  pushbutton_TAP.update();                    // Ler o estado do botão
  if (pushbutton_TAP.getState() == Button::Falling)   // Se pressionado...
    midi.sendCC(85, 127);         // Envia midi CC 85
        }
}
